src/sentry/static/sentry/app/components/asyncComponent.tsx

```diff
@@ -361,30 +379,21 @@ export default class AsyncComponent<
     return [['data', endpoint, this.getEndpointParams()]];
   }
 
-  renderSearchInput({
-    onSearchSubmit,
-    stateKey,
-    url,
-    updateRoute,
-    ...other
-  }: React.ComponentProps<typeof AsyncComponentSearchInput>) {
-    const [firstEndpoint]: any = this.getEndpoints() || [];
+  renderSearchInput({stateKey, url, ...props}: RenderSearchInputArgs) {
+    const [firstEndpoint] = this.getEndpoints() || [null];
     const stateKeyOrDefault = stateKey || (firstEndpoint && firstEndpoint[0]);
     const urlOrDefault = url || (firstEndpoint && firstEndpoint[1]);
     return (
       <AsyncComponentSearchInput
-        updateRoute={updateRoute}
-        onSearchSubmit={onSearchSubmit}
-        stateKey={stateKeyOrDefault}
```

Comment:
Is this intended?

Reply:
Looks fine to me.

Reply:
Yeah not really sure what we were accomplishing here other than maybe making it really explicit?

Reply:
You could potentially pass in a different stateKey, otherwise it uses the first item from `getEndpoints`

Reply:
Ah nevermind, I see that the component doesn't use it

Reply:
Ah I see what you're saying, sorry for some reason I thought you were commenting on the whole red block.

`stateKey` is not a prop on AsyncComponentSearchInput.

Reply:
Looks like these props were here before the TS conversion.